pkg/resource/deploy/step.go

```diff
@@ -600,7 +600,15 @@ func (s *ReadStep) Apply(preview bool) (resource.Status, StepCompleteFunc, error
 			}
 		}
 
+		// If there is no such resource, return an error indicating as such.
+		if result.Outputs == nil {
+			return resource.StatusOK, nil, errors.Errorf("resource '%s' does not exist", id)
```

Comment:
Even though we are failing, we are returning resource.StatusOK - is that Intended?

Reply:
Yes. I don't think we actually use these statuses anywhere, in fact.